### PR TITLE
Sql naming convention topic

### LINKED ISSUE #229: Add a sql naming convention topic and update all sql projects to match
### Link to content

eg : http://syllabus.africacode.net/sql/shop-project/ 

### What don't you like about it?

- Some of the names are redundant. Eg: CustomerID in the Customers table. It should be just "id"
- when a naming convention isn't specified then folks just get creative, better to get into the havit of using standard naming conventions

### Suggestion

1. Make a topic on clean sql (if it does not yet exist) and make sure it describes a sensible naming convention
2. update all sql projects so that the clean sql topic is a pre-requisite

### PR DESCRIPTION
**Pull request by babalwa-mbolekwa**
Related issues: [https://github.com/Umuzi-org/ACN-syllabus/issues/229]

## Description:

What are you up to? Fill us in :)

## I solemnly swear that:

- [x] I ran the hugo server and looked at my changed in the browser with my own eyes
- [x] I ran the linter and there were no errors
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Comment by babalwa-mbolekwa** [comment]
May you please kindly review, I haven't yet added an id and haven't added the topic to other projects as I needed to check if I was doing the work correctly first and if there are any changes requested before I can add it to other SQL projects. Thanks.

File: content/topics/clean-code/sql/_index.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
+_db_id:
```
Comment (sheenarbw) [inline content/topics/clean-code/sql/_index.md]: please remove this line

File: content/topics/clean-code/sql/_index.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+_db_id: 
+content_type: topic
+ready: true
+tags:
+- clean-code
+- sql
+title: Clean Code for SQL
+---
+
+**SQL (Structured Query Language)** is mostly used to manipulate data from a relational database. A database is a container to help organize data and efficiently store and retrieve data. **Relational databases** describe the data and the relationships between data entities.
```
Comment (sheenarbw) [inline content/topics/clean-code/sql/_index.md]: container might be the wrong name. If you google `sql container` or `database container` then all sorts of funny things come up.

File: content/topics/clean-code/sql/_index.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+Remember, good SQL code is easy to read and understand.
+
+### Databases
+If you're creating a school database, name it `school` instead of just `database`
+
+### Tables and table columns
+- Name a table using a singular instead of plural, i.e. `customer`, `account`, e.t.c 
+- Primary key column: It's good practice to name this column as just `id`
+- Foreign key column: Use the specific table's name and a "id" e.g. `customer_id` - that way it easily references the foreign key to the `customer` table
```
Comment (sheenarbw) [inline content/topics/clean-code/sql/_index.md:73]: This is great! 

Can you add an example? Maybe a picture.

**Review by kaleem99 — CHANGES_REQUESTED** [review]
looks good but just one thing needs to be changed, line 13 previously is spelt incorrectly.

**Review by vmonoge — APPROVED** [review]

**Comment by vmonoge** [comment]
The spelling error was corrected

**Review by kaleem99 — APPROVED** [review]

**Review by sheenarbw — CHANGES_REQUESTED** [review]

**Review by Sbonelo01 — APPROVED** [review]

**Review by ry-oc — APPROVED** [review]
changes done :)

**Review by sheenarbw — APPROVED** [review]

**Pull request merged**